api/src/main/java/io/minio/MinioClient.java

```diff
@@ -1138,7 +1139,12 @@ private HttpResponse executeReq(Method method, String region, String bucketName,
           errorXml = scanner.next();
         }
 
-        errorResponse = new ErrorResponse(new StringReader(errorXml));
+        if ("application/xml".equals(response.headers().get("content-type"))) {
+          errorResponse = new ErrorResponse(new StringReader(errorXml));
+        } else {
+          throw new InvalidResponseException();
```

Comment:
Do a negative logic and you don't need `else` part.

Reply:
Modified.